anxcloud/schema_kubernetes.go

```diff
@@ -0,0 +1,138 @@
+package anxcloud
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func schemaKubernetesCluster() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Cluster identifier.",
+		},
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			Description:  "Cluster name.",
+			ForceNew:     true,
+			ValidateFunc: validateLowercaseHostnameRFC1123,
+		},
+		"version": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Kubernetes version.",
+		},
+		"location": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Cluster location.",
+			ForceNew:    true,
+		},
+		"needs_service_vms": {
+			Type:        schema.TypeBool,
+			Description: "Deploy Service VMs providing load balancers and outbound masquerade.",
+			ForceNew:    true,
+			Default:     true,
+			Optional:    true,
+		},
+		"enable_nat_gateways": {
+			Type:        schema.TypeBool,
+			Description: "If enabled, Service VMs are configured as NAT gateways connecting the internal cluster network to the internet.",
+			ForceNew:    true,
+			Default:     true,
+			Optional:    true,
+		},
+		"enable_lbaas": {
+			Type:        schema.TypeBool,
+			Description: "If enabled, Service VMs are set up as LBaaS hosts enabling K8s services of type LoadBalancer.",
+			ForceNew:    true,
+			Default:     true,
+			Optional:    true,
+		},
+	}
+}
+
+func schemaKubernetesNodePool() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Node pool identifier.",
+		},
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "Node pool name.",
+			ValidateFunc: validateLowercaseHostnameRFC1123,
+		},
+		"initial_replicas": {
+			Type:        schema.TypeInt,
+			Required:    true,
+			ForceNew:    true,
+			Description: "Number of initial nodes.",
+		},
+		"cpus": {
+			Type:        schema.TypeInt,
+			Required:    true,
+			ForceNew:    true,
+			Description: "Number of CPUs per node.",
+		},
+		"memory_gib": {
+			Type:        schema.TypeInt,
+			Required:    true,
+			ForceNew:    true,
+			Description: "Memory per node in GiB.",
+		},
+		"disk_size_gib": {
+			Type:        schema.TypeInt,
+			Required:    true,
+			ForceNew:    true,
+			Description: "Disk size per node in GiB.",
+		},
```

Comment:
oh! we have a partly-implemented change pending to support both multiple disks and configure the performance type for them. Maybe it would make sense to have a forward-compatible structure in terraform provider, something like this perhaps?

```hcl
resource "anxcloud_kubernetes_node_pool" "foo" {
    // [...]

    disk {
        size_gib = 20
    }
}
```